File: src/procedures/NavigationProcedure.java
package procedures;

import java.util.ArrayList;

import led.LEDDataBundle;
import led.LEDStripManager;
import led.ProcedureBundleFields;
import procedures.models.navigation.PositionMarker;

public class NavigationProcedure extends Procedure {
	
	public ArrayList<PositionMarker> mMarkers = new ArrayList<PositionMarker>();
	
	public NavigationProcedure(LEDDataBundle _bundle) {
		super(_bundle);
	}

	@Override
	public void start() {
		
	}

	@Override
	void update() {
		for (int i = 0; i < mMarkers.size(); i++) {
			PositionMarker marker = mMarkers.get(i);
			switch (marker.mMarkerSize) {
			case Large:
				
				break;
			case Medium:
				
				break;
			case Small:
				
				break;
			case Single:
				mStrip.setPixel((int)(marker.mXPosition * 180.0f), marker.mColor.toSystemColor());
				break;
			
			default:
				break;
			
			}
		}
		
		finishProcedure();
	}
}

File: src/main/java/procedures/NavigationProcedure.kt
package procedures

import com.github.iamniklas.liocore.led.LEDDataBundle
import com.github.iamniklas.liocore.procedures.Procedure
import procedures.models.navigation.PositionMarker
import procedures.models.navigation.Size

class NavigationProcedure(_bundle: LEDDataBundle) : Procedure(_bundle) {
    var mMarkers = ArrayList<PositionMarker>()
    override fun start() {}
    public override fun update() {
        for (i in mMarkers.indices) {
            val marker = mMarkers[i]
            when (marker.mMarkerSize) {
                Size.Large -> {
                }
                Size.Medium -> {
                }
                Size.Small -> {
                }
                Size.Single -> strip!!.setPixel((marker.mXPosition * 180.0f).toInt(), marker.mColor.toSystemColor())
                else -> {
                }
            }
        }
        finishProcedure()
    }
}